### PR TITLE
[FE] add basic UI

### PR DESCRIPTION
**Pull request by AnCichocka**

### CODE REVIEW

File: frontend/src/components/bars/LevelProgressBar/LevelMiniature.tsx [hunk 1]
```diff
@@ -1,19 +1,26 @@
...
+        level={{
+          name: level.levelName,
+          ordinalNumber: level.ordinalNumber,
+          realLevelNumber: level.ordinalNumber + 1,
+          imageId: level.imageFile?.fileId,
+          minimumPoints: parseFloat(level.minimumPoints),
+          maximumPoints: parseFloat(level.maximumPoints),
+        }}
+      />
     </div>
```
Comment (AnCichocka) [inline frontend/src/components/bars/LevelProgressBar/LevelMiniature.tsx:24]: this component should be removed

File: frontend/src/components/bars/ProgressBar.tsx [hunk 2]
```diff
@@ -12,15 +15,19 @@ export type ProgressBarProps = {
   bounds: { lower: number; upper: number };
   thresholds?: BarThreshold[];
   showPoints?: boolean;
-  label?: string;
+  title?: string;
+  pointsColor?: string;
+  barColor?: string;
```
Comment (AnCichocka) [inline frontend/src/components/bars/ProgressBar.tsx:20]: change to dark/light color

File: frontend/src/components/hallOfFame/HallOfFameMenu.tsx [hunk 3]
```diff
@@ -48,6 +49,6 @@ const styles: Styles = {
     display: "flex",
     gap: 12,
     padding: 12,
-    backgroundColor: "lightgrey",
+    backgroundColor: "grey",
```
Comment (AnCichocka) [inline frontend/src/components/hallOfFame/HallOfFameMenu.tsx:52]: todo token

File: frontend/src/components/images/Avatar.tsx
Comment (AnCichocka) [inline frontend/src/components/images/Avatar.tsx:1]: probably should be removed

File: frontend/src/screens/HallOfFame/HallOfFame.tsx [hunk 4]
```diff
@@ -80,17 +78,18 @@ const styles: Styles = {
...
-    height: "100%",
     display: "flex",
     flexDirection: "column",
+    justifyContent: "center",
   },
   sideBarContainer: {
     display: "flex",
     flexDirection: "column",
-    backgroundColor: "lightblue",
+    backgroundColor: "grey",
```
Comment (AnCichocka) [inline frontend/src/screens/HallOfFame/HallOfFame.tsx:92]: todo token

File: frontend/src/hooks/StudentProfile/useStudentProfileData/useStudentData.tsx [hunk 5]
```diff
@@ -83,15 +87,25 @@ export const useStudentData = (props: {
...
+    loading: headersLoading,
+    error: headersError,
+  } = useCategoriesQuery({ variables: { editionId: editionId as string } });
+
+  const filterHeaderNames: FilterMenuItemm[] =
+    headersData?.categories.map((c) => ({
+      ...c,
+      id: c.categoryId,
+      name: c.categoryName,
+    })) ?? [];
```
Comment (AnCichocka) [inline frontend/src/hooks/StudentProfile/useStudentProfileData/useStudentData.tsx:101]: make it a hook?

File: frontend/src/components/images/Image.tsx [hunk 6]
```diff
@@ -22,6 +56,7 @@ export const Image = ({ size, id, disabled }: ImageProps) => {
 const styles: Styles = {
   img: {
     width: "100%",
+    borderRadius: 12,
```
Comment (AnCichocka) [inline frontend/src/components/images/Image.tsx:59]: this probably should be aligned with img size

File: frontend/src/components/bars/ProgressBar.tsx [hunk 7]
```diff
@@ -46,30 +53,44 @@ export const ProgressBar = ({
...
+            width: `${filledPercent}%`,
+            background:
+              pointsColor && barColor
+                ? `linear-gradient(to right, ${pointsColor}, ${barColor})`
+                : barColor
+                  ? barColor
+                  : "lightblue",
+            borderTopRightRadius: filledPercent === 100 ? BORDER_RADIUS : 0,
+            borderBottomRightRadius: filledPercent === 100 ? BORDER_RADIUS : 0,
+          }}
```
Comment (AnCichocka) [inline frontend/src/components/bars/ProgressBar.tsx:76]: correct

File: frontend/src/components/StudentProfile/table/StudentTable.tsx [hunk 8]
```diff
@@ -83,28 +83,31 @@ export const StudentTable = ({
...
                             editFunctions?.handleDeleteClick(
                               p.points.purePoints?.pointsId,
                             );
                           }
                         }}
-                        isDisabled={
+                        disabled={
                           blockActionButtons || !p.points.purePoints?.pointsId
                         }
+                        color="red"
```
Comment (AnCichocka) [inline frontend/src/components/StudentProfile/table/StudentTable.tsx:110]: token

File: frontend/src/components/StudentProfile/table/FilterMenu/FilterMenuItem.tsx [hunk 9]
```diff
@@ -0,0 +1,39 @@
...
+  lightColor,
+  darkColor,
+  onClick,
+}: FilterMenuItemProps) => {
+  return (
+    <div
+      onClick={onClick}
+      style={{
+        ...styles.item,
+        backgroundColor: isSelected ? darkColor : "transparent",
```
Comment (AnCichocka) [inline frontend/src/components/StudentProfile/table/FilterMenu/FilterMenuItem.tsx:24]: correct

File: frontend/src/components/StudentProfile/table/FilterMenu/FilterMenu.tsx [hunk 10]
```diff
@@ -30,16 +36,13 @@ export default function FilterMenu({
...
+        <FilterMenuItem
+          text={item.name}
+          isSelected={isSelected(item)}
+          lightColor={item.lightColor}
+          darkColor={item.darkColor}
           onClick={() => handleCategoryClick(item)}
-        >
-          {item.name}
-        </div>
+        />
```
Comment (AnCichocka) [inline frontend/src/components/StudentProfile/table/FilterMenu/FilterMenu.tsx:45]: pass item here

File: frontend/src/components/StudentProfile/cards/StudentCard.tsx [hunk 11]
```diff
@@ -1,61 +1,94 @@
...
+        ))}
       </div>
-      <div>prowadzący: {group ? group.teacherDisplayName : "brak"}</div>
 
-      <div style={styles.title}>Course progress</div>
-      <CourseProgressBar totalPoints={totalPoints} />
-    </div>
+      {/* TODO this progress bar probably should be in different card */}
+      {/* <CourseProgressBar totalPoints={totalPoints} title="total progress bar" /> */}
+    </Section>
```
Comment (AnCichocka) [inline frontend/src/components/StudentProfile/cards/StudentCard.tsx:74]: to delete

File: frontend/src/components/StudentProfile/cards/Card/CardItem.tsx
Comment (AnCichocka) [inline frontend/src/components/StudentProfile/cards/Card/CardItem.tsx:1]: rename

File: frontend/src/components/StudentProfile/cards/AnimalCard/AnimalCard.tsx [hunk 12]
```diff
@@ -17,32 +19,49 @@ export const AnimalCard = ({
   nextLevel,
   totalPoints,
 }: AnimalCardProps) => {
+  const items: CardItemProps[] = [
+    { icon: "monster", title: "wielka bestia" },
+    { icon: "level", title: "lvl. 6" },
+    { icon: "level", title: currLevel.grade },
+    { icon: "points", title: "123 - 254" },
```
Comment (AnCichocka) [inline frontend/src/components/StudentProfile/cards/AnimalCard/AnimalCard.tsx:26]: place real data

File: frontend/src/components/IconMapper.tsx [hunk 13]
```diff
@@ -0,0 +1,102 @@
...
+  icon: {
+    width: tokens.padding.m,
+    height: tokens.padding.m,
+    color: tokens.color.text.primary,
+  },
+  clickable: {
+    cursor: "pointer",
+  },
+  disabled: {
+    color: "grey",
```
Comment (AnCichocka) [inline frontend/src/components/IconMapper.tsx:100]: tokens

File: frontend/src/components/Groups/GroupsList/GroupCard.tsx [hunk 14]
```diff
@@ -15,52 +19,64 @@ export const GroupCard = ({
...
+  detailsContainer: {
+    paddingTop: 8,
+    display: "flex",
+    flexDirection: "column",
+    gap: 6,
   },
   editableIndicatorWrapper: {
     top: 140,
-    right: 32,
+    right: 23,
```
Comment (AnCichocka) [inline frontend/src/components/Groups/GroupsList/GroupCard.tsx:79]: xd? 24

File: frontend/src/components/Groups/FilterBar/FilterOptionsSection.tsx [hunk 15]
```diff
@@ -29,51 +33,38 @@ export const FilterOptionsSection = ({
...
+      <div style={styles.optionsContainer}>
+        {options.map((option) => (
+          <div key={option.id} style={styles.optionContainer}>
+            <FormControlLabel
+              control={
+                <Checkbox
+                  checked={selectedIds.includes(option.id)}
+                  onChange={() => handleOptionChange(option.id)}
+                  sx={{
+                    color: tokens.color.text.primary,
```
Comment (AnCichocka) [inline frontend/src/components/Groups/FilterBar/FilterOptionsSection.tsx:46]: move to styles

File: frontend/src/components/FilterButton.tsx [hunk 16]
```diff
@@ -12,22 +14,27 @@ export const FilterButton = ({
...
 };
 
 const styles: Styles = {
   button: {
     padding: 12,
+    paddingTop: 12,
+    paddingBottom: 12,
+    borderRadius: 8,
     cursor: "pointer",
+    transition: "background-color 0.3s ease",
```
Comment (AnCichocka) [inline frontend/src/components/FilterButton.tsx:38]: we need this animation to be exported

File: frontend/src/components/FilterButton.tsx [hunk 17]
```diff
@@ -12,22 +14,27 @@ export const FilterButton = ({
...
-      onClick={onClick}
-    >
-      {option}
+    <div onClick={onClick}>
+      <CustomText
+        size={tokens.font.text}
+        color={isActive ? tokens.color.accent.dark : undefined}
+        style={{
+          ...styles.button,
+        }}
```
Comment (AnCichocka) [inline frontend/src/components/FilterButton.tsx:23]: why

File: frontend/src/components/CustomText.tsx [hunk 18]
```diff
@@ -0,0 +1,39 @@
...
+  color,
+  bold,
+}: CustomTextProps) => {
+  return (
+    <div
+      style={{
+        ...styles.text,
+        fontSize: size ?? tokens.font.text,
+        color: color ?? tokens.color.text.primary,
+        fontWeight: bold ? "bold" : undefined,
```
Comment (AnCichocka) [inline frontend/src/components/CustomText.tsx:26]: change to default props

File: frontend/src/components/CustomIconButton.tsx [hunk 19]
```diff
@@ -0,0 +1,49 @@
...
+    width: 30,
+    height: 30,
+    display: "flex",
+    justifyContent: "center",
+    alignItems: "center",
+    borderRadius: 4,
+    cursor: "pointer",
+  },
+  disabled: {
+    backgroundColor: "grey",
```
Comment (AnCichocka) [inline frontend/src/components/CustomIconButton.tsx:46]: tokens

File: frontend/src/components/CustomIconButton.tsx [hunk 20]
```diff
@@ -0,0 +1,49 @@
...
+  icon,
+  color = "blue",
+  disabled,
+  onClick,
+}: CustomIconButtonProps) => {
+  return (
+    <div
+      style={{
+        ...styles.wrapper,
+        backgroundColor: color ?? "pink",
```
Comment (AnCichocka) [inline frontend/src/components/CustomIconButton.tsx:21]: tokens

File: frontend/src/components/CustomButton.tsx [hunk 21]
```diff
@@ -0,0 +1,60 @@
...
+    paddingBottom: 12,
+    borderRadius: 6,
+    cursor: "pointer",
+    display: "inline-block",
+  },
+  disabled: {
+    cursor: "auto",
+    backgroundColor: "grey",
+  },
+  hoovered: {
```
Comment (AnCichocka) [inline frontend/src/components/CustomButton.tsx:57]: to delete

File: frontend/src/components/CustomButton.tsx [hunk 22]
```diff
@@ -0,0 +1,60 @@
...
+    padding: 12,
+    paddingTop: 12,
+    paddingBottom: 12,
+    borderRadius: 6,
+    cursor: "pointer",
+    display: "inline-block",
+  },
+  disabled: {
+    cursor: "auto",
+    backgroundColor: "grey",
```
Comment (AnCichocka) [inline frontend/src/components/CustomButton.tsx:55]: tokens

File: frontend/src/components/CustomButton.tsx [hunk 23]
```diff
@@ -0,0 +1,60 @@
...
+  children: string;
+  disabled?: boolean;
+};
+
+export const CustomButton = ({
+  onClick,
+  children,
+  disabled,
+}: CustomButtonProps) => {
+  const [isHovered, setIsHovered] = useState(false);
```
Comment (AnCichocka) [inline frontend/src/components/CustomButton.tsx:17]: to delete

**Review by Szyntos — APPROVED** [review]
butiful

**Pull request merged**